src/asynch.rs

```diff
@@ -40,14 +44,29 @@ where
 {
     /// Create a new TLS connection with the provided context and a async I/O implementation
     ///
-    /// NOTE: The record buffer should be sized to fit an encrypted TLS record and the TLS handshake
-    /// record. The maximum value of a TLS record is 16 kB, which should be a safe value to use.
-    pub fn new(delegate: Socket, record_buf: &'a mut [u8]) -> Self {
+    /// NOTE: The record read buffer should be sized to fit an encrypted TLS record. The size of this record
+    /// depends on the server configuration, but the maximum allowed value for a TLS record is 16 kB, which
+    /// should be a safe value to use.
+    ///
+    /// The write record buffer can be smaller than the read buffer. During write [`TLS_RECORD_OVERHEAD`] over overhead
+    /// is added per record, so the buffer must at least be this large. Large writes are split into multiple records if
+    /// depending on the size of the write buffer.
+    /// The largest of the two buffers will be used to encode the TLS handshake record, hence either of the
+    /// buffers must at least be large enough to encode a handshake.
+    pub fn new(
+        delegate: Socket,
+        record_read_buf: &'a mut [u8],
+        record_write_buf: &'a mut [u8],
+    ) -> Self {
```

Comment:
We should add an assertion here that record_write_buf.len() >= TLS_RECORD_OVERHEAD.

Reply:
Likewise, at least one of them should fit the minimal size of the handshake, but it's a bit more complex to determine so can be left out for now.

Reply:
I added the assertion